lib/tlscontext.c

```diff
@@ -848,6 +854,50 @@ tls_context_setup_sigalgs(TLSContext *self)
   return TRUE;
 }
 
+#if SYSLOG_NG_HAVE_DECL_SSL_CONF_CMD
+static gboolean
+_apply_cmd_context(TLSContext *self)
+{
+  gboolean result = TRUE;
+
+  if (self->conf_cmds_list) {
+    for (GList *next_pair = self->conf_cmds_list; next_pair; next_pair = next_pair->next)
+      {
+        gchar *cmd = next_pair->data;
+        next_pair = next_pair->next;
+        gchar *value = next_pair->data;
+
+        int add_result = SSL_CONF_cmd(self->ssl_conf_ctx, cmd, value);
+        if (add_result < 1)
+          msg_error("Error setting up SSL conf context, invalid cmd or value in config list",
+                    evt_tag_str("cmd", cmd),
+                    evt_tag_str("value", value),
+                    tls_context_format_location_tag(self));
+        result = result && (add_result >= 1);
+      }
+  }
+  return result;
+} 
+#endif
+
+static gboolean
+tls_context_setup_cmd_context(TLSContext *self)
+{
+#if SYSLOG_NG_HAVE_DECL_SSL_CONF_CMD
+  self->ssl_conf_ctx = SSL_CONF_CTX_new();
+
+  const int ctx_flags = SSL_CONF_FLAG_FILE | SSL_CONF_FLAG_CLIENT | SSL_CONF_FLAG_SERVER | SSL_CONF_FLAG_CERTIFICATE | SSL_CONF_FLAG_SHOW_ERRORS;
+  int set_flags = SSL_CONF_CTX_set_flags(self->ssl_conf_ctx, ctx_flags);
+  g_assert(set_flags == ctx_flags && "Could not set SSL CONF CTX flags");
```

Comment:
do we need this assert here? shouldn't we handle as errors instead of an abort() ?

Reply:
I just followed the practice I found e.g. in tls_context_setup_ocsp_stapling, but you are right, it could be an error message too now (as this setup is not in the constructor anymore like was during my earlier versions)

Reply:
ok